### ISSUE TITLE
Show all table rows on first page load when using reloadCallback 

### ISSUE THREAD

**Issue by chrisvel** [Zizzamia/ng-tasty#205]
In your example http://zizzamia.com/ng-tasty/directive/table-server-side/reload you can reload the table when the reloadCallback is called. But when the table is initially displayed, the object isn't fetched with a GET request, but only with the reloadCallback. How can I trigger the table load on initial page display ?

**Comment by pgouv** [comment]
Call search() maybe ? 
Rotary

**Comment by chrisvel** [comment]
`search()` isn't mandatory, I am calling `$scope.reloadCallback();` from another function. Somehow `bind-reload` overrides `bind-resource-callback` when declared in `<table>` tag.

**Comment by Zizzamia** [comment]
The purpuse of the `reload` callback, it's to have the full control of when the table should be loaded.
http://zizzamia.com/ng-tasty/directive/table-server-side/reload

@chrisvel just for better understanding, your error happen only when you declare everything in the `<table>` tag?
If you do it as show in the documentation should work?

If it is the case, I have to investigate to understand what is the bug.
Thanks

**Comment by chrisvel** [comment]
@Zizzamia yes like your example. I don't think it's a bug, there might be a workaround. Imagine listing a table with all elements and then try to filter the main object in order to exclude some. The way it works now is that it shows nothing until a GET request is trigger when the callback is called.

**Comment by chrisvel** [comment]
@Zizzamia I found a workaround but I am not sure how much safe it is. 

```
var callInit = function() {
      $scope.reloadCallback();
    }

angular.element(document).ready(callInit);
```

Please advice and If you agree you could add it to the documentation. 

EDIT: This works only on page load and not when moving between pages. Logically, 

```
var callInit = function() {
      $scope.reloadCallback();
    }

callInit();
```

should be enough but it doesn't work. Any help will be greatly appreciated because I am using it in a big project right now and have been stuck for several days.

**Comment by chrisvel** [comment]
FYI http://stackoverflow.com/questions/34839674/how-to-trigger-the-table-load-on-every-page-view-with-angularjs-and-ngtasty

**Comment by piyushpatil027** [comment]
Hi Zizzamia ,

Hope you are doing great.I am using (http://zizzamia.com/ng-tasty/directive/table-server-side/reload )  library. When the table is initially displayed, the object isn't fetched with a GET request, but only with the reloadCallback. How can I trigger the table load on initial page display ?

I am getting same problem in my code.How can this problem sovle?

**Comment by masoodtrb** [comment]
@piyushpatil027 +1

**Comment by newbro** [comment]
I am also facing this problem. Actually the default bind-resource-callback + bind-filters already serve my needs (I too wish to load up all rows in table onload, and add filtering options on UI). However I spot there are some minor issues with the way it watches filter variable (debouncing problem?) which results in inconsistent search data so I end up using bind-reload so I can limit the ajax call. I am pretty much done, except I need a way to load all the data into the table on page load unlike the example.

BTW thanks for the work on this, seems to be a much cleaner solution than using the angular-datatable hack.

[UPDATE]
Just found out I can actually override the default tableConfig constant by using bind-theme (the naming doesn't really help here just my opinion)

so using bind-theme="customTheme" and $scope.customTheme = { loadOnInit:true } was able to solve my problem.

**Comment by olegderecha** [comment]
> so using bind-theme="customTheme" and $scope.customTheme = { loadOnInit:true } was able to solve my problem.

Excellent, thank you! 
I set **bind-theme="{loadOnInit: true}"** in markup.

**Comment by HesyRa** [comment]
> I set bind-theme="{loadOnInit: true}" in markup.

Ty, also worked for me!! gj

**Comment by Zizzamia** [comment]
Sorry guys, had very intense 6 months with Plan (https://getplan.co), spending most of my time with Angular 2 recently. 

Here guys the documentation about `bind-theme` http://zizzamia.com/ng-tasty/directive/table/custom-theme, you can override several default property.

Also @newbro, what name you suggest? Open to any suggestion :)
